examples/advanced/Tutorial3/simulation/FairTestDetectorGeo.cxx

```diff
@@ -26,13 +28,19 @@ const char* FairTestDetectorGeo::getModuleName(Int_t m)
         ASCII file should start with FairTestDetector otherwise they will
         not be constructed
     */
-    sprintf(modName, "torino%i", m + 1);
+    int result_length = snprintf(modName, maxbuf-1, "torino%i", m + 1);
+    if (!(result_length > 0 && result_length < static_cast<int>(maxbuf))) {
+      LOG(fatal) << "Buffer overrun in snprintf.";
+    }
```

Comment:
There should be no negative `m`. The function is only called from one place in the old ASCII geometry interface in a loop which runs from 0 to the maximum number of detectors. But your are right we should not rely on that. The m+1 comes from the fact that in C++ array counting starts at 0 but the name of the detector in the text file has normal counting which starts with 1. 

I put a fatal in first place to see if we have any problems at all. In this special example I think since these are hardcoded values one should fix the buffer size in the class if there is a problem which is very unlikely. the value of m would have to be a number with 19 characters which would be possible but I would see this as an error if it would appear.
So in my opinion I would see a fatal here really as an error and nothing one should work around.

The hard coded torino id probably a leftover and one could and should use here the data member. I will look to that.
Probably the code duplication in all the classes can be moved to an implementation in the base class.

Reply:
> There should be no negative `m`.

Let's add an `assert(m >= 0);` to the beginning of the function body to document this assumption.

> The function is only called from one place in the old ASCII geometry interface in a loop which runs from 0 to the maximum number of detectors. (...) The m+1 comes from the fact that in C++ array counting starts at 0 but the name of the detector in the text file has normal counting which starts with 1.

I would love a comment explaining this shifted index in the generated name.

> Probably the code duplication in all the classes can be moved to an implementation in the base class.

If you decide to refactor this code (which I would appreciate): Are we sure we need this optimization of formatting into an existing buffer?

If not, let's consider using something simple like `std::format("{}{}", myDetector->GetName(), m);` (with a fallback to fmtlib - which we ship in FairSoft since many years - in case the `<format>` std library is not available).